lib/engine/step/g_18_zoo/special_track.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require_relative '../special_track'
+
+module Engine
+  module Step
+    module G18ZOO
+      class SpecialTrack < SpecialTrack
+        def lay_tile(action, _extra_cost: 0, _entity: nil, _spender: nil)
+          hex = action.hex
+          tile = action.tile
+
+          if %w[O M MM].include?(hex.location_name) &&
+            tile.color == :yellow &&
+            !tile.name.end_with?('(water)') && !tile.name.end_with?('(mountain)')
+            tile = case tile.name + '_' + hex.location_name
+                   when '7_O'
+                     Engine::Game::G18ZOO::TILE_W7.dup
```

Comment:
this looks copy and pasted...

Reply:
I need the same feature from track also here; I'm using a "super" at the end, and when I tried to put all of this inside a module I lost the "super" reference; I'll try to improve this anyway